src/gfn/actions.py

```diff
@@ -34,6 +34,10 @@ def __init__(self, tensor: TT["batch_shape", "action_shape"]):
             tensor: tensor of actions
         """
         self.tensor = tensor
+        assert len(tensor.shape) > len(self.action_shape), (
```

Comment:
I know it's pedantic but I'd add the comment

`# Ensure the tensor has all action dimensions and at least one batch dimension.`